dftimewolf/lib/exporters/timesketch_tam.py

```diff
@@ -0,0 +1,250 @@
+    timelines = sketch.list_timelines()
+    while True:
+      if all(tl.status in ['fail', 'ready', 'timeout', 'archived']
+             for tl in timelines):
+        break
+      time.sleep(10)
+
+  def _GetSketchIDFromAttributes(self) -> int:
+    """Attempts to retrieve a Timesketch ID from ticket attributes.
+
+    Returns:
+      int: the sketch idenifier, or 0 if one was not available.
+    """
+    attributes = self.state.GetContainers(containers.TicketAttribute)
+    for attribute in attributes:
+      if attribute.name == self._SKETCH_ATTRIBUTE_NAME:
+        sketch_match = re.search(r'sketch/(\d+)/', attribute.value)
+        if sketch_match:
+          sketch_id = int(sketch_match.group(1), 10)
+          return sketch_id
+    return 0
+
+  def Process(self, container: containers.File) -> None:
+    """Executes a Timesketch export."""
+
+    recipe_name = self.state.recipe.get('name', 'no_recipe')
+
+    description = container.name
+    if description:
+      name = description.rpartition('.')[0]
+      name = name if name else container.name
+      name = name.replace(' ', '_').replace('-', '_')
+      timeline_name = '{0:s}_{1:s}'.format(
+          recipe_name, name)
+    else:
+      timeline_name = recipe_name
+
+    with importer.ImportStreamer() as streamer:
+      streamer.set_sketch(self.sketch)
+      streamer.set_timeline_name(timeline_name)
+
+      path = container.path
+      streamer.add_file(path)
+      if streamer.response and container.description:
+        streamer.timeline.description = container.description
+
+    api_root = self.sketch.api.api_root
+    self.host_url = api_root.partition('api/v1')[0]
+    sketch_url = '{0:s}sketches/{1:d}/'.format(self.host_url, self.sketch.id)
+    message = 'Your Timesketch URL is: {0:s}'.format(sketch_url)
+    report_container = containers.Report(
+        module_name='TimesketchExporterThreaded',
+        text=message,
+        text_format='markdown')
+    self.state.StoreContainer(report_container)
+
+    if self.wait_for_timelines:
+      self.logger.info('Waiting for timelines to finish processing...')
+      self._WaitForTimelines()
+
+    for analyzer in self._analyzers:
+      results = self.sketch.run_analyzer(
+          analyzer_name=analyzer, timeline_name=timeline_name)
+      if not results:
+        self.logger.info('Analyzer [{0:s}] not able to run on {1:s}'.format(
+            analyzer, timeline_name))
+        continue
+      session_id = results.id
+      if not session_id:
+        self.logger.info(
+            'Analyzer [{0:s}] didn\'t provide any session data'.format(
+                analyzer))
+        continue
+      self.logger.info('Analyzer: {0:s} is running, session ID: {1:d}'.format(
+          analyzer, session_id))
+      self.logger.info(results.status_string)
+
+  @staticmethod
+  def GetThreadOnContainerType() -> Type[interface.AttributeContainer]:
+    return containers.File
+
+  def GetThreadPoolSize(self) -> int:
```

Comment:
Is this really worth to have as a method? Since the returned value is always the same, might as well have a constant? Or is this an override of the parent class? In which case just ignore me

Reply:
Yep this is a method to override from the parent class. The value return needs to be determined by the implementation - Though to be fair, I put no effort into working out what the "correct" value here should be.